metrics/sources/kubelet/kubelet.go

```diff
@@ -233,25 +233,30 @@ metricloop:
 	return metricSetKey, cMetrics
 }
 
-func (this *kubeletMetricsSource) ScrapeMetrics(start, end time.Time) *DataBatch {
+func (this *kubeletMetricsSource) ScrapeMetrics(start, end time.Time) (*DataBatch, error) {
 	containers, err := this.scrapeKubelet(this.kubeletClient, this.host, start, end)
+
 	if err != nil {
 		glog.Errorf("error while getting containers from Kubelet %s: %v", this.host, err)
```

Comment:
remove this line, we'll log it later on.  Just return an error like `return fmt.Errorf("unable to scrape Kubelet for container metrics: %v", err)`